source/userguide/active_learning.rst

```diff
@@ -0,0 +1,269 @@
+Active Learning in MindMeld
+===========================
+
+Once a conversational application is deployed, capturing user logs can be highly beneficial to understand new trends and diversity in terms of how users are phrasing their queries for different intents. These logs can improve the quality of the training data and overall classifier performance. However, manually analyzing these logs and annotating them with the correct domain, intent and entity labels is costly and time consuming. Moreover, as the number of users for the system increase, the number of logs to be annotated also increases by many folds. This reduces the scalability of the manual annotation process. For such scenarios, we can use Active Learning.
+
+
+What is Active Learning?
+^^^^^^^^^^^^^^^^^^^^^^^^
+In this section, we take a closer look at what active learning is and how it works within MindMeld.
+
+Active Learning is an approach to continuously and iteratively pick data that is most informative for the models to train on from a pool of data-points. By choosing these informative data-points, active learning provides a higher chance of improving the accuracy of the model with fewer training examples. If the pool of data-points is unannotated, active learning also greatly reduces the number of queries that need manual annotation. MindMeld provides this inbuilt functionality to select these must-have queries from existing data or additional logs and datasets to get the best out of your conversational applications.
+
+MindMeld's NLP pipeline consists of a heirarchy of classifier and tagger models for domain, intent, entity and role classification. Each classifier takes as input the query text and extracted features to generate a probability distribution for the corresponding classes in the heirarchy. These probability distributions can be strong indicator of the classifier's confidence in its predictions and can be useful in determining whether the query needs further reflection. 
+
+Say for a query A from the logs, the selected class (or the highest probability class) in the domain classifier is at 0.95 or 95% and the remaining 5% is distributed amongst the rest of the domains, this could potentially indicate a high confidence by the classifier in its prediction. Whereas a query B has a probability 0.5 for the selected class, possibly indicating lower prediction confidence. Based on these values, the active learning strategies will select query B over query A for adding back to the training data. With query B now in the training data with proper annotation, the system has a higher chance of performing better on similar confusable queries. Later sections explain how different strategies use these confidence scores to select important queries.
```

Comment:
```suggestion
Say for a query A from the logs, the classifier assigns a probability of 0.95 or 95% to the selected class (i.e, the highest probability class) in the domain classifier and the remaining 5% is distributed amongst the rest of the domains. This indicates a high confidence by the classifier in its prediction. Whereas, for a query B, if the classifier assigns a probability 0.5 for the selected class, it would indicate a lower confidence in its prediction. Based on these confidence values, certain active learning strategies will select query B over query A which can be annotated and added to the training data. Once the classifier is trained with this additional data, it has a higher chance of performing better on similar confusable queries. Later sections explain how different strategies use these confidence scores to select important queries.
```